web/checks_api.go

```diff
@@ -100,17 +122,38 @@ func ApiCreateChecksCatalogHandler(s services.ChecksService) gin.HandlerFunc {
 // @Success 200 {object} JSONChecksSettings
 // @Failure 404 {object} map[string]string
 // @Router /api/checks/{id}/settings [get]
-func ApiCheckGetSettingsByIdHandler(s services.ChecksService) gin.HandlerFunc {
+func ApiCheckGetSettingsByIdHandler(consul consul.Client, s services.ChecksService) gin.HandlerFunc {
 	return func(c *gin.Context) {
-		id := c.Param("id")
+		clusterId := c.Param("id")
 
-		selectedChecks, err := s.GetSelectedChecksById(id)
+		// TODO: this has absolutely to be refactored once we've got the hosts service
+		clusters, err := cluster.Load(consul)
 		if err != nil {
-			_ = c.Error(NotFoundError("could not find check selection"))
+			_ = c.Error(err)
 			return
 		}
 
-		connSettings, err := s.GetConnectionSettingsById(id)
+		clusterItem, ok := clusters[clusterId]
+		if !ok {
+			_ = c.Error(NotFoundError("could not find cluster"))
+			return
+		}
+
+		filterQuery := fmt.Sprintf("Meta[\"trento-ha-cluster-id\"] == \"%s\"", clusterId)
+		hosts, err := hosts.Load(consul, filterQuery, nil)
+		if err != nil {
+			_ = c.Error(err)
+			return
+		}
+
+		nodes := NewNodes(s, clusterItem, hosts)
+
+		selectedChecks, err := s.GetSelectedChecksById(clusterId)
+		if err != nil {
+			selectedChecks = models.SelectedChecks{SelectedChecks: pq.StringArray{}}
```

Comment:
Why have you `hidden` this error here?
I understand that you did this because if the `GetSelectedChecksById` is used when no checks are selected, it fails.
But I guess the same will happen with the `GetConnectionSettingsById` call.

In any case, I don't think you need to initialize a new `models.SelectedChecks` here. The `GetSelectedChecksById` already returns an empty struct of those in case of failure. This would space the import of `pq`

Reply:
:thinking: I think you're right, I'll work on this.